### PR TITLE
pixman: fixing aarch64-darwin build

### PR DESCRIPTION
**Pull request by vcunat**
https://hydra.nixos.org/build/197830077

###### Things done

- Built on platform(s)
  - [ ] no rebuild x86_64-linux
  - [ ] no rebuild aarch64-linux
  - [ ] no rebuild x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
@ofBorg build pixman

**Comment by vcunat** [comment]
I doesn't work :-(

**Comment by vcunat** [comment]
I suppose in the worst case we could split aarch64-darwin away, using the older version with security patch (we patch on 22.05).

**Comment by sternenseemann** [comment]
Probably because `gccStdenv` has this:

```
      # Remove libcxx/libcxxabi, and add clang for AS if on darwin (it uses
      # clang's internal assembler).
      extraBuildInputs = lib.optional stdenv.hostPlatform.isDarwin clang.cc;
```

*If* the gcc assembler can't work on darwin, it seems like we would need to downgrade.

**Comment by vcunat** [comment]
@ofBorg build pixman

**Comment by sternenseemann** [comment]
FWIW the `gccStdenv` stuff was put together in 2017 (including using clang for `AS`), so who knows how it should be today: a5168fe4570acaed5ff306789be5d038f4b95662

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19291,7 +19291,13 @@ with pkgs;
 
   atkmm_2_36 = callPackage ../development/libraries/atkmm/2.36.nix { };
 
-  pixman = callPackage ../development/libraries/pixman { };
+  pixman = callPackage ../development/libraries/pixman (
+    # https://gitlab.freedesktop.org/pixman/pixman/-/issues/46
+    lib.optionalAttrs stdenv.hostPlatform.isAarch64 {
+      # Just gccStdenv won't work, as on aarch64-darwin it still uses clang's assembler.
+      stdenv = stdenv.override { cc = buildPackages.gcc; };
```
Comment (sternenseemann) [inline pkgs/top-level/all-packages.nix:19298]: This should take care of the requisites problem:

```suggestion
      stdenv = overrideCC stdenv buildPackages.gcc;
```

**Comment by vcunat** [comment]
@ofBorg build pixman

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/top-level/all-packages.nix:19298]: Hmm right, thanks.  Sounds better than the more complex variant I wrote.

**Comment by vcunat** [comment]
Still doesn't seem to work: https://logs.nix.ci/?key=nixos/nixpkgs.200156&attempt_id=2cc36828-c17d-42ed-84cc-2c87f908a937

**Comment by vcunat** [comment]
I haven't tried this one:
@ofBorg build pixman

**Comment by vcunat** [comment]
But really... someone actually interested in aarch64-darwin should be solving such problems.

**Comment by vcunat** [comment]
https://logs.nix.ci/?key=nixos/nixpkgs.200156&attempt_id=8beea391-ccbf-476c-be0f-e6351f6cadeb
I guess I'm giving up 🤷🏽

**Comment by reckenrode** [comment]
I was able to get it to build by disabling ARM64 Neon on aarch64-darwin. This is using the default, clang-based `stdenv`.

```diff
diff --git a/pkgs/development/libraries/pixman/default.nix b/pkgs/development/libraries/pixman/default.nix
index 49f5fbe8815..39753ac5492 100644
--- a/pkgs/development/libraries/pixman/default.nix
+++ b/pkgs/development/libraries/pixman/default.nix
@@ -18,7 +18,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ libpng ];
 
-  configureFlags = lib.optional stdenv.isAarch32 "--disable-arm-iwmmxt";
+  configureFlags = lib.optional stdenv.isAarch32 "--disable-arm-iwmmxt"
+   ++ lib.optional (stdenv.isAarch64 && stdenv.isDarwin) "--disable-arm-a64-neon";
 
   preConfigure = ''
     # https://gitlab.freedesktop.org/pixman/pixman/-/issues/62
```

**Comment by vcunat** [comment]
@ofBorg build pixman

**Comment by vcunat** [comment]
Thanks.  OfBorg confirmed success, so I pushed it there with you as author.

**Pull request merged**